streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java

```diff
@@ -148,8 +150,12 @@ default void init(final StateStoreContext context, final StateStore root) {
     default <R> QueryResult<R> query(
         Query<R> query,
         PositionBound positionBound,
-        boolean collectExecutionInfo) {
+        QueryConfig config) {
```

Comment:
This is a public API change. I will need to update the KIP. We need this so that we can skip the cache for example for windowed queries in ksql